mikeio/dfs.py

```diff
@@ -58,15 +59,15 @@ def read(self, items=None, time_steps=None):
         """
         self._open()
 
-        items, item_numbers, time_steps = self._get_valid_items_and_timesteps(
-            items, time_steps
-        )
+        item_numbers = valid_item_numbers(self._dfs.ItemInfo, items)
+        n_items = len(item_numbers)
+
+        time_steps = valid_timesteps(self._dfs.FileInfo, time_steps)
 
         for t in time_steps:
```

Comment:
Is this still necessary?

Reply:
I assume that you're referring to the lines below:

        for t in time_steps:
            if t > (self.n_timesteps - 1):
                raise IndexError(f"Timestep {t} is > {self.n_timesteps-1}")

which I have now deleted